engine/src/graphics/device.cpp

```diff
@@ -173,6 +178,14 @@ namespace gen
 		m_graphicsQueue = m_device->getQueue(m_gpu.queueFamily, 0);
 	}
 
+	void GraphicsDevice::createCommandPoolAndBuffer()
+	{
+		m_commandPool = m_device->createCommandPoolUnique(vk::CommandPoolCreateInfo(vk::CommandPoolCreateFlags(), m_gpu.queueFamily));
+
+		m_commandBuffer =
+			std::move(m_device->allocateCommandBuffersUnique(vk::CommandBufferAllocateInfo(m_commandPool.get(), vk::CommandBufferLevel::ePrimary, 1)).front());
```

Comment:
VulkanHPP offers overloads for such functions that take a pointer and count instead of returning a vector. It's not a big deal here as this only happens once, but later on we'll be creating command buffers on the fly for various stuff, where it will add needless allocation overhead.